extensions/wikia/AdEngine/js/lookup/amazonMatch.js

```diff
@@ -139,26 +148,49 @@ define('ext.wikia.adEngine.lookup.amazonMatch', [
 		return amazonCalled;
 	}
 
+	function isValidSlot(slotName, validSlotNames) {
+		// No Object.keys on IE8
+		checkIfObjectKeysNotSupported('isValidSlot()', false);
+
+		var isValid = false;
+
+		Object.keys(validSlotNames).forEach(function (k) {
+			var validSlotName = validSlotNames[k];
+
+			if (!isValid) {
+				isValid = slotName === validSlotName;
+			}
+		});
+
+		return isValid;
+	}
+
+	function checkIfObjectKeysNotSupported(functionName, retval) {
```

Comment:
Since this module is useless without Object.keys anyways, let's return the full interface, but without any implementation if there's no Object.keys and then assume Object.keys is supported throughout the module.